google-cloud-bigquery/lib/google/cloud/bigquery/schema.rb

```diff
@@ -276,141 +244,86 @@ def date name, description: nil, mode: :nullable
         #   end
         #
         def record name, description: nil, mode: nil
+          # TODO: do we need to fail if no block was given?
```

Comment:
FWIW: If we are going to leave the block check then we might as well ensure the record has a nested field. I am fine with leaving as is until we discuss with @quartzmo.